include/dd/Package.hpp

```diff
@@ -1675,17 +1687,27 @@ namespace dd {
                     edge[idx] = ResultEdge::zero;
                     for (auto k = 0U; k < rows; k++) {
                         LEdge e1{};
+                        auto  element1 = rows * i + k;
                         if (!x.isTerminal() && x.p->v == var) {
-                            e1 = x.p->e[rows * i + k];
+                            e1 = x.p->e[element1];
                         } else {
+                            // Effectively inserts an identity node
                             e1 = xCopy;
+                            if ((element1 == 1 or element1 == 2) and (std::is_same_v<LeftOperandNode, mNode> or std::is_same_v<LeftOperandNode, dNode>)) {
+                                e1.w = Complex::zero;
```

Comment:
I believe that this can be improved. This is not really generic and efficient. Same applied to the statement below:
```c++
if constexpr (std::is_same_v<LeftOperandNode, vNode>) {
  e1 = xCopy;
} else {
  if (i == k) {
    e1 = xCopy;
  } else {
    e1 = Ledge::zero;
}
```

Note that this could potentially be improved even further. As of now, The left operand node can never be a vector DD. That should probably be a `static_assert` somewhere within this method. So the first (constexpr) `if` condition is technically not even necessary here.
Furthermore, we don't support skipped nodes for vector DDs at the moment. This limitation could be made more explicit.